plugins/modules/zos_blockinfile.py

```diff
@@ -574,42 +570,30 @@ def main():
     # state=present, insert/replace a block with matching regex pattern
     # state=absent, delete blocks with matching regex pattern
     if parsed_args.get('state') == 'present':
-        return_content = present(src, block, quotedString(marker), quotedString(ins_aft), quotedString(ins_bef), encoding, force)
+        return_content = present(src, block, marker, ins_aft, ins_bef, encoding, force)
     else:
-        return_content = absent(src, quotedString(marker), encoding, force)
+        return_content = absent(src, marker, encoding, force)
     stdout = return_content.stdout_response
     stderr = return_content.stderr_response
     rc = return_content.rc
+    stdout = stdout.replace('/d', '\\\\d')
     try:
-        # change the return string to be loadable by json.loads()
-        stdout = stdout.replace('/c\\', '/c\\\\')
-        stdout = stdout.replace('/a\\', '/a\\\\')
-        stdout = stdout.replace('/i\\', '/i\\\\')
-        stdout = stdout.replace('$ a\\', '$ a\\\\')
-        stdout = stdout.replace('1 i\\', '1 i\\\\')
-        if block:
-            stdout = stdout.replace(block, quoted_string_output_json(block))
-        if ins_aft:
-            stdout = stdout.replace(ins_aft, quoted_string_output_json(ins_aft))
-        if ins_bef:
-            stdout = stdout.replace(ins_bef, quoted_string_output_json(ins_bef))
         # Try to extract information from stdout
-        ret = json.loads(stdout)
-        ret['cmd'] = ret['cmd'].replace("u'", '"')
-
-        result['cmd'] = ret['cmd']
-        result['changed'] = ret['changed']
-        result['found'] = ret['found']
-        # Only return 'rc' if stderr is not empty to not fail the playbook run in a nomatch case
-        # That information will be given with 'changed' and 'found'
-        if len(stderr):
-            result['stderr'] = str(stderr)
-            result['rc'] = rc
+        ret = json.loads("""{0}""".format(stdout))
```

Comment:
I don't think there is a need to use a multiline string and format here. We could just use `json.loads(stdout)`?

Reply:
In this case as we aren't escaping any special character the """ """ allow us to pass values as _ / ' to the python API so is required

Reply:
Got it! would be nice to have it as a comment